### DISCUSSION TITLE
What determines the type of a callback style watch

### DISCUSSION THREAD

**Discussion by JPStrydom** [react-hook-form/react-hook-form#9358] [Q&A] [answered]
When using a `watch` in a callback style:
```
watch((values, { name, type }) => console.log(values, name, type));
```
What determines the values for `type`? I've only ever seen it be `'change'` or `undefined` and [the documentation](https://react-hook-form.com/api/useform/watch) doesn't mention it at all.

I've also notice that the second parameter on the watch callback (the object that we destruct `name` and `type` from) sometimes contains a `values` property as well, which isn't always the same as the `values` in the first parameter. What's that about?

**Comment by bluebill1049** [comment, marked as answer]
`type` is returned by the input event type.

**Reply by JPStrydom** [reply]
Is there any way to set this when calling `setValue`?

**Reply by bluebill1049** [reply]
not at the moment, but maybe we can consider that, do you want to send a feature request for that? I probably should support reset or any other way of value update as well.

**Reply by JPStrydom** [reply]
@bluebill1049 I made a [feature request](https://github.com/react-hook-form/react-hook-form/issues/9363) for this. Please let me know if it suffices.
